### PR TITLE
Add Kubernetes Locally

### PR DESCRIPTION
**Pull request by jdgarvey**
Kubernetes has been added locally. Set the image pull policy to "Never" so it just uses local images for now. Will start pushing them to a registry in another PR.

Introduced both local and release versions of the client and server. Local versions use volumes to facilitate hot-reloading, release versions leverage multi-stage builds to make the final images as lightweight as possible

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+indent_size = 2
+insert_final_newline = true
+trim_trailing_whitespace = true
+
+[*.md]
+max_line_length = off
+trim_trailing_whitespace = false
+
+[Makefile]
+indent_style = tab
```
Comment (VicAv99) [inline .editorconfig]: We ALL need this in life 😂

File: Makefile [hunk 2]
```diff
@@ -1,5 +1,5 @@
-start: ## Builds and starts the project in a docker container
-	@docker-compose up --remove-orphans --build
+install-node-modules: ## Install dependencies locally
```
Comment (VicAv99) [inline Makefile]: Can we name this just `install`?

File: client/libs/core-data/src/lib/users/users.service.ts [hunk 3]
```diff
@@ -1,11 +1,13 @@
...
+
 import { Injectable } from '@angular/core';
 import { HttpClient } from '@angular/common/http';
 
 import { Observable } from 'rxjs';
 
 import { User } from './user.model';
 
-const BASE_URL = 'http://localhost:8080/api/v1';
+const BASE_URL = `${process.env.ENDPOINT}/api/v1`;
```
Comment (johncblandii) [inline client/libs/core-data/src/lib/users/users.service.ts]: Angular won't have access to `process.env.ENDPOINT` at runtime.

File: Makefile (hunk 2 above)
Comment (jdgarvey) [reply Makefile]: We can, I just wanted to point out that it is not doing anything on the backend.

File: client/libs/core-data/src/lib/users/users.service.ts (hunk 3 above)
Comment (jdgarvey) [reply client/libs/core-data/src/lib/users/users.service.ts]: Correct...but it does at build time thanks to a custom Angular builder that leverages webpack.
This is the tut I followed: https://medium.com/@fidelisclayton/system-environment-variables-in-angular-1f4a922c7b4c

File: client/Dockerfile-release [hunk 4]
```diff
@@ -0,0 +1,27 @@
+# Build the base image from the source code
+FROM node:10-alpine as build
```
[the remark this answers is not in the document]
Comment (jdgarvey) [reply client/Dockerfile-release]: Fixed

File: client/Dockerfile-release [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+COPY . .
+
+ARG ENDPOINT=http://localhost:31003
+
+RUN echo "Building release image" && \
+  ENDPOINT=$ENDPOINT yarn build --configuration=production
+
+# Create the final image
+FROM nginx:1.17.3-alpine
```
[the remark this answers is not in the document]
Comment (jdgarvey) [reply client/Dockerfile-release]: Fixed

**Pull request merged**